src/charm.py

```diff
@@ -157,6 +157,18 @@ def create_attach_config(token, services=None):
 @retry(ProcessExecutionError)
 def attach_subscription(token, env, services=None):
     """Attach an ubuntu-advantage subscription using the specified token and services."""
+    # First detach if already attached
```

Comment:
Instead of the comment, maybe change the docstring above to say something like "Ensures the Ubuntu Pro attachment to using the specified token and services, detaching first in case the system is already attached".

I would also rename this to just `attach`.